lib/decorators/EventPropertiesDecorator.hpp

```diff
@@ -114,6 +114,9 @@ namespace MAT_NS_BEGIN {
                 record.data.push_back(data);
             }
 
+            if (eventProperties.GetTimestamp() != 0)
```

Comment:
Is the timestamp guaranteed to be 0 when not set? Would it make sense to define this as a `const` variable in the header and compare against that?

Reply:
The underlying value is an int64_t, so by definition if it's unset it's 0